packages/flutter/lib/src/material/data_table.dart

```diff
@@ -1047,6 +1082,52 @@ class DataTable extends StatelessWidget {
   }
 }
 
+/// Height settings for DataRows.
+/// Either a fixed row height or a row height that adjusts to its content can be specified.
+@immutable
+class DataTableRowHeight with Diagnosticable {
+  /// Creates height settings where all rows are created with the same fixed height.
+  /// [fixedHeight] defaults to kMinInteractiveDimension to adhere to the Material Design specifications.
+  const DataTableRowHeight.fixed({
+    double? height = kMinInteractiveDimension
+  }) : fixedHeight = height, topBottomPadding = 0.0;
+
+  /// Creates height settings where the row height adjusts to the content.
+  const DataTableRowHeight.contentBased({
+    required this.topBottomPadding,
+  }) : fixedHeight = null;
+
+  /// Fixed height value.
+  /// If set then all rows will geht this fixed height, if null then content based height is used.
+  final double? fixedHeight;
+
+  /// The minimum padding between the content of each data cell and the top and bottom of the row.
+  /// If fixed height is used then this value is set to 0.0 because the content is vertically centered which makes padding unneeded.
```

Comment:
why not make it null instead?

Reply:
For me it is slightly more intuitive to set verticalPadding to 0.0 (instead null) if fixedHeight is specified because 0.0 is the padding that is applied. In other words, if the fixedHeight is 100 and the content height is also 100 then the content would exactly fit into the cell. So there would be no padding.